pom.xml

```diff
@@ -118,8 +154,17 @@
                 <plugin>
                     <groupId>org.jenkins-ci.tools</groupId>
                     <artifactId>maven-hpi-plugin</artifactId>
+                    <extensions>true</extensions>
                     <configuration>
-                        <pluginFirstClassLoader>true</pluginFirstClassLoader>
+                        <pluginFirstClassLoader>false</pluginFirstClassLoader>
```

Comment:
This require more thorough testing than that. As I said, this manifested as runtime problems in unpredictable places as I expect this would break other use-cases then this one.

https://issues.jenkins-ci.org/browse/JENKINS-39259
https://issues.jenkins-ci.org/browse/JENKINS-39282